### PR TITLE
Add time data to issue report if test is annotated with @Stopwatch

### LINKED ISSUE #689: Create interaction between `@Stopwatch` and `@Issue`
If a test is annotated with both `@Issue` and `@Stopwatch`, the test's run time could be included in the information published for each test.

### PR DESCRIPTION
**Pull request by Michael1993**
Closes #689 

Proposed commit message:

```
Create interaction between @Stopwatch and @Issue (#689 / #743)

If a test is annotated with both @Issue and @Stopwatch, the test's
run time is included in the `IssueTestCase`.

Closes: #689
PR: #743
```

---
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done. 

Documentation (general)
* [ ] There is documentation (Javadoc and site documentation; added or updated)
* [ ] There is implementation information to describe _why_ a non-obvious source code / solution got implemented
* [ ] Site documentation has its own `.adoc` file in the `docs` folder, e.g. `docs/report-entries.adoc`
* [ ] Site documentation in `.adoc` file references demo in `src/demo/java` instead of containing code blocks as text
* [ ] Only one sentence per line (especially in `.adoc` files)
* [ ] Javadoc uses formal style, while sites documentation may use informal style

Documentation (new extension)
* [ ] The `docs/docs-nav.yml` navigation has an entry for the new extension
* [ ] The `package-info.java` contains information about the new extension

Code
* [ ] Code adheres to code style, naming conventions etc.
* [ ] Successful tests cover all changes
* [ ] There are checks which validate correct / false usage / configuration of a functionality and there are tests to verify those checks
* [ ] Tests use [AssertJ](https://assertj.github.io/doc/) or our own [PioneerAssert](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#assertions) (which are based on AssertJ)

Contributing
* [ ] A prepared commit message exists
* [ ] The list of contributions inside `README.md` mentions the new contribution (real name optional)

### CODE REVIEW

**Comment by Bukama** [comment]
Note: Draft for #689

File: src/main/java/org/junitpioneer/jupiter/IssueTestCase.java [hunk 1]
```diff
@@ -61,23 +65,36 @@ public Status result() {
...
 	@Override
 	public int hashCode() {
-		return Objects.hash(testId, result);
+		return Objects.hash(testId, result, elapsedTime);
 	}
 
 	@Override
 	public String toString() {
+		if (Objects.nonNull(elapsedTime)) {
+			return "IssueTestCase{" + "uniqueName='" + testId + '\'' + ", result='" + result + '\'' + ", elapsedTime='"
```
Comment (Bukama) [inline src/main/java/org/junitpioneer/jupiter/IssueTestCase.java:95]: Why double all the code and not just add the additional time information when it's present?

**Comment by Bukama** [comment]
> > Could you do me a favor and check if the Junit test case data also contains it's execution duration (I'm not sure by mind)? If it's available we should discuss (in the issue): Do we want to store the time from the `@Stopwatch` extension or do we want to store the JUnit build in time.
> 
> Not sure what you mean by that?

I had in mind that the `TestExecutionResult` we access in the `IssueExtensionExecutionListener` already has information about the duration of the test execution, but I was wrong. You only get the status and possible `Throwable`.

**Review by Bukama — APPROVED** [review]

**Review by nipafx — APPROVED** [review]
Still missing a commit message but look good otherwise. My 👍🏾 for a merge.

**Pull request merged**